view/frontend/web/js/view/checkout/email.js

```diff
@@ -1,40 +1,61 @@
 define([
     'uiComponent',
     'jquery',
-    'jquery/ui'
+    'jquery/ui',
+    '!domReady'
 ], function(Component) {
     'use strict';
+    // initialize the customerData prior to returning the component
+    var _klaviyoCustomerData = window.customerData;
 
     return Component.extend({
         initialize: function () {
             this._super();
-            this.bindEmailListener();
+            this._klaviyoCustomerData = _klaviyoCustomerData;
+            this._email;
+            this.handleCheckout();
             return this;
         },
-
+        handleCheckout: function() {
+          if (this.isUserLoggedIn() && this._email){
+            this.postUserEmail(this._email);
+          } else {
+            this.bindEmailListener();
+          }
+        },
+        isUserLoggedIn: function () {
+          this._email = this._klaviyoCustomerData ? this._klaviyoCustomerData.email : undefined;
+          if (this._email) {
+            return true;
+          }
+        },
         bindEmailListener: function() {
+            // jquery overrides this, so let's create an instance of the parent
+            var self = this;
             console.log('Klaviyo_Reclaim - Binding to #customer-email');
             jQuery('#maincontent').delegate('#customer-email', 'change', function(event) {
-                var customer_email = jQuery(this).val();
-
-                var path = window.location.pathname;
-                if (path.slice(-1) == '/' ) {
-                    path = path.slice(0, -1);
-                }
+                self._email = jQuery(this).val();
+                self.postUserEmail(self._email);
+            });
+        },
+        postUserEmail: function (customer_email) {
+          var path = window.location.pathname;
+          if (path.slice(-1) == '/' ) {
+              path = path.slice(0, -1);
+          }
 
-                var url = window.location.protocol + '//' + window.location.host + path.substring(0,path.lastIndexOf("/"));
+          var url = window.location.protocol + '//' + window.location.host + path.substring(0,path.lastIndexOf("/"));
```

Comment:
why all the slicing and substring then appending, can we just do all this manipulation as part of one line instead of 3 lines? whatever it is is only used once in that ajax call as far as I can see

Reply:
looks like it's some cleaning, i don't know if i have a recommendation to do this in 1 line as it would get even more bulky - open to suggestions though!